src/app/static/src/app/store/plotSelections/utils.ts

```diff
@@ -0,0 +1,145 @@
+import { Commit } from "vuex";
+import { CalibrateMetadataResponse, FilterOption, FilterRef, FilterTypes, PlotSettingOption } from "../../generated";
+import { PlotName } from "./plotSelections";
+import { PlotSelectionUpdate, PlotSelectionsMutations } from "./mutations";
+import { RootState } from "../../root";
+
+export const filtersAfterUseShapeRegions = (filterTypes: FilterTypes[], rootState: RootState) => {
+    const filters = [...filterTypes];
+    const area = filters.find(f => f.id == "area");
+    if (area && area.use_shape_regions) {
+        const regions: FilterOption[] = rootState.baseline.shape!.filters!.regions ?
+            [rootState.baseline.shape!.filters!.regions] : [];
+
+        const index = filters.findIndex(f => f.id === "area");
+        const { use_shape_regions, ...areaFilterConfig } = area;
+        filters[index] = { ...areaFilterConfig, options: regions };
+    }
+    return filters;
+};
+
+type NestedFilterOption = {
+    id: string
+    label: string,
+    children?: NestedFilterOption[]
+}
+
+class Q {
+    items: NestedFilterOption[]
+    constructor() { this.items = [] }
+    enqueue(item: NestedFilterOption) { this.items.push(item) }
+    dequeue() { return this.items.pop()! }
+    isEmpty() { return this.items.length === 0 }
+}
+
+const getFullNestedFilters = (filterOptions: NestedFilterOption[]) => {
+    const fullFilterOptions: NestedFilterOption[] = [];
+    const q = new Q();
+    filterOptions.forEach(op => q.enqueue(op));
+    while (!q.isEmpty()) {
+        const currentOption = q.dequeue();
+        fullFilterOptions.push(currentOption);
+        if (currentOption.children && currentOption.children.length > 0) {
+            currentOption.children.forEach(op => q.enqueue(op));
+        }
+    }
+    return fullFilterOptions;
+};
+
+export const filtersInfoFromPlotSettings = (
```

Comment:
Does this need to be exported?

Reply:
Maybe for testing, we should write a test before we merge this as it's fairly complicated but I think fine to leave that until we've settled on how we're representing it like you say.

Reply:
i thought it might be useful to reuse this for the changing plot controls mutations that is coming next!